Ginger/GingerCoreNET/AnalyzerLib/AnalyzerUtils.cs

```diff
@@ -109,84 +109,93 @@ public List<string> RunBusinessFlowAnalyzer(BusinessFlow businessFlow, Observabl
+                                }
+
+                            }
 
-                    foreach (var variableIssueGroup in missingVariableIssuesGroupList)
+                        }
+
+                        List<string> tempList = AnalyzeAction.GetUsedVariableFromAction(action);
+                        MergeVariablesList(usedVariablesInActivity, tempList);
+                    });
+
+                    List<string> activityVarList = AnalyzeActivity.GetUsedVariableFromActivity(activity);
+
+                    MergeVariablesList(usedVariablesInActivity, activityVarList);
+                    ReportUnusedVariables(activity, usedVariablesInActivity, issuesList);
+                    MergeVariablesList(usedVariablesInBF, usedVariablesInActivity);
+
+
+                    usedVariablesInActivity.Clear();
+                });
+
+                //Get all the missing variable issues Grouped by Variable name
+                lock (mMissingVariableIssueLock)
+                {
+                    if (missingVariableIssueList.Count != 0)
                     {
-                        //If for specific variable, all the issues are for set variable action then we support Auto Fix
-                        var canAutoFix = variableIssueGroup.All(x => x is AnalyzeAction && ((AnalyzeAction)x).mAction.GetType() == typeof(ActSetVariableValue));
+                        var missingVariableIssuesGroupList = missingVariableIssueList.GroupBy(x => x.IssueReferenceObject);
 
-                        if (canAutoFix)
+                        foreach (var variableIssueGroup in missingVariableIssuesGroupList)
                         {
-                            foreach (AnalyzeAction issue in variableIssueGroup)
+                            //If for specific variable, all the issues are for set variable action then we support Auto Fix
+                            var canAutoFix = variableIssueGroup.All(x => x is AnalyzeAction && ((AnalyzeAction)x).mAction.GetType() == typeof(ActSetVariableValue));
+
+                            if (canAutoFix)
                             {
-                                issue.CanAutoFix = AnalyzerItemBase.eCanFix.Yes;
-                                issue.FixItHandler += MarkSetVariableActionAsInactive;
+                                foreach (AnalyzeAction issue in variableIssueGroup)
+                                {
+                                    issue.CanAutoFix = AnalyzerItemBase.eCanFix.Yes;
+                                    issue.FixItHandler += MarkSetVariableActionAsInactive;
+                                }
                             }
-                        }
 
+                        }
                     }
+
                 }
-           
-            }
 
-            ReportUnusedVariables(businessFlow, usedVariablesInBF, issuesList);
+                ReportUnusedVariables(businessFlow, usedVariablesInBF, issuesList);
+            }
+            catch (Exception ex) 
```

Comment:
with this we are supressing that exception. so if there is exception in the first flow we will not analyzer remaining and also the user will not know that there is exception unless debug mode is set. what is the exception ? can he handle the reason for the exception instead of try catch ?